c-lightning/2021-08-09-developer-call.md

```diff
@@ -0,0 +1,195 @@
+
+I am working on adding [offer](https://bolt12.org/) support on c-lightning REST and then planning to add experimental offers functionality on the [RTL](https://github.com/Ride-The-Lightning/RTL) UI as well. Thinking through the UX and the different scenarios on how to handle offers, both generate offers and if somebody pays an offer how to set up a recurring payment, what type of UX. I’m working on the backend right now, c-lightning REST and then figuring out an experimental UI for offers. 
+
+Interestingly Nadav (shesek) is working on integrating with [Spark](https://github.com/shesek/spark-wallet). He’s hit the same kind of problems. His initial release which will come out soon, he is waiting for the final c-lightning release, when that comes out it won’t support recurrence. All the big UX questions are around the recurrence stuff. He is going to have the basic offers and he is hoping in the next release he will get recurrence. It requires local storage, have timers and things like that that are a whole new set of UX challenges. Obviously that is what everyone wants.
+
+RTL does not have a database. To support recurring payments for offers we have to create a database and create a program which will wake up, look up the outstanding offers to be paid and then go back to sleep. This kind of logic will be completely new in RTL to handle this.
+
+He is wrestling with the same thing. We have a [datastore plugin](https://github.com/ElementsProject/lightning/pull/4674) now for c-lightning that is going to be built in in future, there is a plugin for older versions. It is just a key, value store for plugins to use. In particular Spark will be using it for storing the recurring payments. It would be nice to have that standardized so that whatever front end you were using would use the same thing. It is still very early days. I have implemented the plugin for the datastore and I have implemented the API as a pull request, it will be in the next release. The key is a list so it is a slightly structured key, it is a JSON list, you can have levels effectively, a hierarchy. But very simple functionality, replace, append, atomic replace and list. It was a compromise, I don’t want to store everything but a lot of plugins want data. The [summary plugin](https://github.com/lightningd/plugins/tree/master/summary) even, it keeps uptime data. They are all stashing it in their little databases. It is nice to have it in one so you can take that database with you and migrate. You don’t want to have to go “If you are using this plugin you also have to pull this other file”. Standardizing how we store that information is definitely something that I would like. I will start an email chain with the three of us at least on that.
```

Comment:
```suggestion
He is wrestling with the same thing. We have a [datastore plugin](https://github.com/ElementsProject/lightning/pull/4674) now for c-lightning that is going to be built in future, there is a plugin for older versions. It is just a key, value store for plugins to use. In particular Spark will be using it for storing the recurring payments. It would be nice to have that standardized so that whatever front end you were using would use the same thing. It is still very early days. I have implemented the plugin for the datastore and I have implemented the API as a pull request, it will be in the next release. The key is a list so it is a slightly structured key, it is a JSON list, you can have levels effectively, a hierarchy. But very simple functionality, replace, append, atomic replace and list. It was a compromise, I don’t want to store everything but a lot of plugins want data. The [summary plugin](https://github.com/lightningd/plugins/tree/master/summary) even, it keeps uptime data. They are all stashing it in their little databases. It is nice to have it in one so you can take that database with you and migrate. You don’t want to have to go “If you are using this plugin you also have to pull this other file”. Standardizing how we store that information is definitely something that I would like. I will start an email chain with the three of us at least on that.
```

Reply:
Done